database/db.go

```diff
@@ -0,0 +1,85 @@
+package database
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"sync"
+
+	_ "github.com/lib/pq"
+
+	"durn2.0/conf"
+	"durn2.0/durn"
+)
+
+var mutex sync.Mutex
+var db *sql.DB
+
+// CreateDBConnection Initializes the database connection using
+// the connection details specified in env-vars
+func CreateDBConnection() {
+	c := conf.ReadConfiguration()
+	psqlconn := fmt.Sprintf("host=localhost port=%d user='%s' password='%s' dbname='%s' sslmode=disable", c.DBPort, c.DBUser, c.DBPassword, c.DBName)
+
+	mutex.Lock()
+	defer mutex.Unlock()
+	var err error
+	db, err = sql.Open("postgres", psqlconn)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func DisconnectDB() {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	db.Close()
+}
+
+func QueryAllVoters() ([]durn.Voter, error) {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	query := `SELECT * FROM valid_voters`
+	rows, err := db.Query(query)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	var res []durn.Voter
+
+	for rows.Next() {
+		var username durn.Voter
+		err = rows.Scan(&username)
+		res = append(res, username)
+	}
+
+	return res, nil
+}
+
+func InsertElection(e durn.Election) error {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	query := `INSERT INTO Elections(id, name, published, finalized, opentime, closetime) VALUES ($1, $2, $3, $4, $5, $6)`
```

Comment:
Sorry about looking at this so late, I know it is kind of time critical. But what is the reasoning behind doing raw SQL rather than using an ORM system like [`gorm`](https://gorm.io/)? 

Think it would look more like:
```go
election := Election{Name: e.Name, Published: e.IsOpen, Finalized: e.IsFinalized, OpenTime: e.OpenTime, CloseTime: e.CloseTime}
result := db.Create(&election)
```

and models, migrations, transactions, and stuff would be handled automatically?

Reply:
Using something like this sound very reasonable, I will take a look
It seems much nicer